crates/levm/src/vm.rs

```diff
@@ -25,6 +26,46 @@ impl VM {
                     self.stack.push(value_to_push);
                     self.increment_pc_by(32);
                 }
+                Opcode::MLOAD => {
+                    // spend_gas(3);
+                    let offset = self.stack.pop().unwrap().try_into().unwrap();
+                    let value = self.memory.load(offset);
+                    self.stack.push(value);
+                }
+                Opcode::MSTORE => {
+                    // spend_gas(3);
+                    let offset = self.stack.pop().unwrap().try_into().unwrap();
+                    let value = self.stack.pop().unwrap();
+                    let mut value_bytes = [0u8; 32];
+                    value.to_big_endian(&mut value_bytes);
+
+                    self.memory.store_bytes(offset, &value_bytes);
+                }
+                Opcode::MSTORE8 => {
+                    // spend_gas(3);
+                    let offset = self.stack.pop().unwrap().try_into().unwrap();
+                    let value = self.stack.pop().unwrap();
+                    let mut value_bytes = [0u8; 32];
+                    value.to_big_endian(&mut value_bytes);
+
+                    self.memory
+                        .store_bytes(offset, value_bytes[31..32].as_ref());
+                }
+                Opcode::MSIZE => {
+                    // spend_gas(2);
+                    self.stack.push(self.memory.size());
+                }
+                Opcode::MCOPY => {
+                    // spend_gas(3) + dynamic gas
+                    let dest_offset = self.stack.pop().unwrap().try_into().unwrap();
+                    let src_offset = self.stack.pop().unwrap().try_into().unwrap();
+                    let size = self.stack.pop().unwrap().try_into().unwrap();
+                    if size == 0 {
+                        continue;
+                    }
+
+                    self.memory.copy(src_offset, dest_offset, size);
+                }
```

Comment:
The `.pop().unwraps()` are probably not a good idea. A malicious actor could crash the service by giving you any of these without pushing the required argument. These should return an error (and panic in the EVM sense of the word, that is revert and consume all gas) when the stack is empty.
The `.try_into().unwrap()` are probably replaceable by a call to `as` or by just keeping the low bits.

Reply:
You're right! We're tracking this and will handle it [here](https://github.com/lambdaclass/ethereum_rust/issues/539).

Reply:
Yeah the `unwraps` are never a good idea, I wanted to add them to expose what can panic as much as possible, and fix them with proper error handling in the EVM later as I am not that sure what our aproach to handling the errors will be!